packages/polyfill/source/CharacterData.ts

```diff
@@ -16,7 +15,13 @@ export class CharacterData extends ChildNode {
       str = typeof data === 'string' ? data : String(data);
     }
     this[DATA] = str;
-    hooks.setText?.(this as any, str);
+    if (this[OWNER_DOCUMENT]) {
+      this[OWNER_DOCUMENT].defaultView[HOOKS].setText?.(this as any, str);
+    } else {
+      setTimeout(() =>
```

Comment:
This is an issue because of us calling `setData` in the constructor, right? Instead of doing this, could we just assign to `this[DATA]` directly in the constructor (applying the same stringifying logic implemented above), which avoids this triggering a hook call?

Reply:
I considered that but that would skip the initial hook call and change the existing behaviour. I'm ok with that but it's a change in the existing behaviour. Alternatively, we can manually set the value via `this[DATA]` and schedule the hook call within the constructor.

Reply:
I am OK to break this behavior. I think the library is actually broken in a different way here — we expose a `createText` hook https://github.com/Shopify/remote-dom/blob/ce50f4bf86f59cbf0a33cfa33144d47125131a3e/packages/polyfill/source/hooks.ts#L10, which should be used to cover the "initial value of a text node" case, but we don't seem to call that hook anywhere 😬

Reply:
Whoops haha Should I add `setTimeout(() => this[HOOKS]?.createText ...)` here then?